test/server/lib/DocApi.ts

```diff
@@ -854,13 +854,36 @@ function testDocApi() {
     async function getColumnFieldsMapById(url: string) {
       const result = await axios.get(url, chimpy);
       assert.equal(result.status, 200);
-      return new Map(
+      return new Map<string, object>(
           result.data.columns.map(
             ({id, fields}: {id: string, fields: object}) => [id, fields]
           )
       );
     }
 
+    async function checkPut(
+      columns: [RecordWithStringId, ...RecordWithStringId[]],
+      params: Record<string, any>,
+      expectedFieldsByColId: Record<string, object>,
+    ) {
+      const {url} = await generateDocAndUrl();
+      const body: ColumnsPut = { columns };
+      const resp = await axios.put(url, body, {...chimpy, params});
+      assert.equal(resp.status, 200);
+      console.log("resp.data = ", resp.data);
+      const fieldsByColId = await getColumnFieldsMapById(url);
```

Comment:
It's possible that if this returned an object instead of a map then the right method such as https://www.chaijs.com/api/assert/#method_deepnestedinclude could significantly simplify this code, but I'm not sure.

Reply:
I attempted some changes, and it seems like you cannot say something like `A, B and C should exist, whatever their values are` like this way:

```ts
      await checkPut([COLUMN_TO_ADD], {}, {
        A: {}, B: {}, C: {}, 'Foo.label': COLUMN_TO_ADD.fields.label, 'Foo.id': COLUMN_TO_ADD.id
      });
```

Here are the changes I made so far (I can commit to let you check or experiment by yourself):
```diff
diff --git a/test/server/lib/DocApi.ts b/test/server/lib/DocApi.ts
index f9f1879f..bf04eec5 100644
--- a/test/server/lib/DocApi.ts
+++ b/test/server/lib/DocApi.ts
@@ -854,17 +854,18 @@ function testDocApi() {
     async function getColumnFieldsMapById(url: string) {
       const result = await axios.get(url, chimpy);
       assert.equal(result.status, 200);
-      return new Map<string, object>(
-          result.data.columns.map(
-            ({id, fields}: {id: string, fields: object}) => [id, fields]
-          )
-      );
+      const retValue: Record<string, object> = {};
+      const columns = result.data.columns as Array<{ id: string, fields: object }>;
+      for (const {id, fields} of columns) {
+        retValue[id] = fields;
+      }
+      return retValue;
     }
 
     async function checkPut(
       columns: [RecordWithStringId, ...RecordWithStringId[]],
       params: Record<string, any>,
-      expectedFieldsByColId: Record<string, object>,
+      expectedFieldsByColId: Record<string, any>,
     ) {
       const {url} = await generateDocAndUrl();
       const body: ColumnsPut = { columns };
@@ -873,15 +874,7 @@ function testDocApi() {
       console.log("resp.data = ", resp.data);
       const fieldsByColId = await getColumnFieldsMapById(url);
 
-      assert.deepEqual(
-        [...fieldsByColId.keys()],
-        Object.keys(expectedFieldsByColId),
-        "The updated table should have the expected columns"
-      );
-
-      for (const [colId, expectedFields] of Object.entries(expectedFieldsByColId)) {
-        assert.deepInclude(fieldsByColId.get(colId), expectedFields);
-      }
+      assert.deepNestedInclude(fieldsByColId, expectedFieldsByColId);
     }
 
     const COLUMN_TO_ADD = {
@@ -902,7 +895,7 @@ function testDocApi() {
 
     it('should create new columns', async function () {
       await checkPut([COLUMN_TO_ADD], {}, {
-        A: {}, B: {}, C: {}, Foo: COLUMN_TO_ADD.fields
+        A: {}, B: {}, C: {}, 'Foo.label': COLUMN_TO_ADD.fields.label, 'Foo.id': COLUMN_TO_ADD.id
       });
     });
 
```